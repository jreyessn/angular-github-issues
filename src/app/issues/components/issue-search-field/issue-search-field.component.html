<mat-form-field class="w-full">
    <mat-label>Buscar</mat-label>
    <input 
        [formControl]="searchInputControl"
        matInput 
        placeholder="Username, Repository, Title" autocomplete="off" >
</mat-form-field>