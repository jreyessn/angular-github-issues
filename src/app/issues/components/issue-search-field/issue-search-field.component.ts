import { Component, EventEmitter, OnInit, Output } from '@angular/core';
import { FormControl } from '@angular/forms';
import { debounceTime } from 'rxjs';

@Component({
  selector: 'app-issue-search-field',
  templateUrl: './issue-search-field.component.html',
  styleUrls: ['./issue-search-field.component.scss']
})
export class IssueSearchFieldComponent implements OnInit {

  @Output() search: EventEmitter<string> = new EventEmitter()

  /**
   * Search form control
   */
   searchInputControl: FormControl = new FormControl();

  constructor() { }

  ngOnInit(): void {
    this.searchInputControl.valueChanges.pipe(
          debounceTime(500)
        )
        .subscribe((value) => 
          this.search.emit(value)
        )
  }

}
