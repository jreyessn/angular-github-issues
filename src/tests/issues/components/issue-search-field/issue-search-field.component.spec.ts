import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatSelectModule } from '@angular/material/select';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatTooltipModule } from '@angular/material/tooltip';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { IssueSearchFieldComponent } from '../../../../app/issues/components/issue-search-field/issue-search-field.component';

describe('IssueSearchFieldComponent', () => {
  let component: IssueSearchFieldComponent;
  let fixture: ComponentFixture<IssueSearchFieldComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ IssueSearchFieldComponent ],
      imports: [
        MatToolbarModule,
        MatIconModule,
        MatButtonModule,
        MatInputModule,
        MatTooltipModule,
        MatProgressSpinnerModule,
        MatSelectModule,
        FormsModule,
        ReactiveFormsModule,
        BrowserAnimationsModule
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(IssueSearchFieldComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  
  it('Debe hacer el match con el snapshot', () => {
    expect(compiled).toMatchSnapshot()
  });

  // it()

});
