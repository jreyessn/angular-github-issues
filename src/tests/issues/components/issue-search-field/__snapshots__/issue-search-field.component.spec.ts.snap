// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSearchFieldComponent Debe hacer el match con el snapshot 1`] = `
<div
  id="root1"
>
  <mat-form-field
    class="mat-form-field w-full mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <input
            aria-invalid="false"
            aria-required="false"
            autocomplete="off"
            class="mat-input-element mat-form-field-autofill-control ng-untouched ng-pristine ng-valid cdk-text-field-autofill-monitored"
            data-placeholder="Username, Repository, Title"
            matinput=""
          />
          <span
            class="mat-form-field-label-wrapper"
          >
            <label
              class="mat-form-field-label mat-empty mat-form-field-empty"
            >
              
              <mat-label>
                Buscar
              </mat-label>
              
              
            </label>
            
          </span>
        </div>
        
      </div>
      <div
        class="mat-form-field-underline"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      
      <div
        class="mat-form-field-subscript-wrapper"
      >
        
        <div
          class="mat-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
        
      </div>
    </div>
  </mat-form-field>
</div>
`;
